bigtable/internal/transport/connpool_test.go

```diff
@@ -114,131 +202,285 @@ func setupTestServer(t *testing.T, service *fakeService) string {
 	return lis.Addr().String()
 }
 
+func dialBigtableserver(addr string) (*BigtableConn, error) {
+	conn, err := grpc.Dial(addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return nil, err
+	}
+	return NewBigtableConn(conn, "test-instance", "test-profile"), nil
+}
+
+func dialBigtableserverWithInstanceNameAndAppProfile(addr string, instanceName, appProfile string) (*BigtableConn, error) {
+	conn, err := grpc.Dial(addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return nil, err
+	}
+	return NewBigtableConn(conn, instanceName, appProfile), nil
+}
+
 func TestSelectRoundRobin(t *testing.T) {
-	pool := &BigtableChannelPool{}
+	pool := &BigtableChannelPool{rrIndex: 0}
 
 	// Test empty pool
-	idx, err := pool.selectRoundRobin()
-	if idx != -1 {
-		t.Errorf("selectRoundRobin on empty pool got index %d, want -1", idx)
-	}
-	if err == nil {
-		t.Errorf("selectRoundRobin on empty pool got nil error, want non-nil")
+	entry, err := pool.selectRoundRobin()
+	if entry != nil {
+		t.Errorf("selectRoundRobin on empty pool got entry, want nil")
 	}
-	if err != errNoConnections {
+	if !errors.Is(err, errNoConnections) {
 		t.Errorf("selectRoundRobin on empty pool got error %v, want %v", err, errNoConnections)
 	}
 
 	// Test single connection pool
-	pool.conns = make([]*grpc.ClientConn, 1)
-	pool.load = make([]int64, 1)
-	idx, err = pool.selectRoundRobin()
-	if idx != 0 {
-		t.Errorf("selectRoundRobin on single conn pool got index %d, want 0", idx)
+	pool.conns.Store([]*connEntry{{}})
+	entry, err = pool.selectRoundRobin()
+	if entry == nil {
+		t.Errorf("selectRoundRobin on single conn pool got nil entry")
 	}
 	if err != nil {
 		t.Errorf("selectRoundRobin on single conn pool got error %v, want nil", err)
 	}
 
 	// Test multiple connections
 	poolSize := 3
-	pool.conns = make([]*grpc.ClientConn, poolSize)
-	pool.load = make([]int64, poolSize)
+	conns := make([]*connEntry, poolSize)
+	for i := 0; i < poolSize; i++ {
+		conns[i] = &connEntry{}
+	}
+	pool.conns.Store(conns)
 	pool.rrIndex = 0
 
-	// Test wrapping around
 	for i := 0; i < poolSize*2; i++ {
 		expectedIdx := i % poolSize
-		idx, err = pool.selectRoundRobin()
-		if idx != expectedIdx {
-			t.Errorf("selectRoundRobin call %d got index %d, want %d", i+1, idx, expectedIdx)
-		}
+		entry, err = pool.selectRoundRobin()
 		if err != nil {
 			t.Errorf("selectRoundRobin call %d got error %v, want nil", i+1, err)
+			continue
+		}
+		if entry != conns[expectedIdx] {
+			t.Errorf("selectRoundRobin call %d got entry for index %d, want %d", i+1, entryIndex(conns, entry), expectedIdx)
 		}
 	}
 }
 
-func TestSelectLeastLoadedRandomOfTwo(t *testing.T) {
-	pool := &BigtableChannelPool{}
-
-	// Test empty pool
```

Comment:
No action needed. But for the next time:
This test file's diff on GitHub is pretty hard to review, because it shows as if functions were replaced but they are not. I got a much saner diff locally using `--diff-algorithm=patience` but that's extra moves/efforts.
Maybe better to keep comments like this, because then the diff algorithm would have additional "anchors" to produce more readable diff.